src/main/ruby/core/dir_glob.rb

```diff
@@ -337,8 +338,7 @@ def self.glob(pattern, flags, matches=[])
       # Rubygems uses Dir[] as a glorified File.exist?  to check for multiple
       # extensions. So we went ahead and sped up that specific case.
 
-      if flags == 0 and
-             m = /^([a-zA-Z0-9_.\-\/\s]*[a-zA-Z0-9_.])(?:\{([^{}\/\*\?]*)\})?$/.match(pattern)
+      if flags == 0 && m = TRAILING_BRACES.match(pattern)
```

Comment:
Should be `and` if there is an assignment, otherwise swapping those conditions would break.

Reply:
This is what I get for combining my two least favorite features of Ruby.